File: LAB_1/Parrot.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Parrot {
    private List<String> phrases;
    private Random random;

    public Parrot(String initialPhrase) {
        phrases = new ArrayList<>();
        phrases.add(initialPhrase);
        random = new Random();
    }

    public void say(int times) {
        if (!phrases.isEmpty()) {
            for (int i = 0; i < times; i++) {
                int randomIndex = random.nextInt(phrases.size());
                System.out.println(phrases.get(randomIndex));
            }
        } else {
            System.out.println("I don't have anything to say!");
        }
    }

    public void addPhrase(String newPhrase) {
        phrases.add(newPhrase);
    }

    public static void main(String[] args) {
        Parrot parrot = new Parrot("It's rather late!");

        parrot.say(3); 
        parrot.addPhrase("Oh well!");
        parrot.addPhrase("Whatever!");
        parrot.say(2); 
    }
}

File: LAB_1/Parrot.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cstdlib>
#include <ctime>

using namespace std;

class Parrot {
private:
    vector<string> phrases;

public:
    Parrot() {
        srand(time(0));  
    }

    void addPhrase(string newPhrase) {
        phrases.push_back(newPhrase);
    }

    void say(int times) {
        for (int i=0; i<times; i++) {
                if (phrases.empty()) {
                    cout << "I don't have anything to say" << endl;
                }
                else {
                int randomIndex = rand() % phrases.size();
                cout << phrases[randomIndex] << endl;
                }
            }
         
        
    }
};

int main() {
    Parrot parrot;
    parrot.say(2);
    parrot.addPhrase("Hello!");
    parrot.say(1);
    parrot.addPhrase("I'm tired!");
    parrot.addPhrase("To the left!");
    parrot.say(2);  

    return 0;
}